File: src/main/java/io/github/richardson/Font.java
package io.github.richardson;

import processing.core.PApplet;
import processing.core.PFont;

public class Font {
    private final PApplet p;
    private final PFont font;


    Font(PApplet p, String path, float size) {
        this.p = p;
        font = p.createFont(path, size, true);
    }

    public void apply() {
        p.textFont(font);
    }
}

File: src/main/kotlin/io/github/richardson/Font.kt
package io.github.richardson

import processing.core.PApplet
import processing.core.PFont

class Font internal constructor(private val p: PApplet, path: String?, size: Float) {
    private val font: PFont?


    init {
        font = p.createFont(path, size, true)
    }

    fun apply() {
        p.textFont(font)
    }
}
